tasks/gitHelpers.ts

```diff
@@ -0,0 +1,197 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { spawnSync } from 'child_process';
+import * as fs from 'fs';
+import * as path from 'path';
+import { Octokit } from '@octokit/rest';
+import { NugetPackageInfo } from './offlinePackagingTasks';
+import { PlatformInformation } from '../src/shared/platform';
+import { platformSpecificPackages } from './offlinePackagingTasks';
+
+export interface GitOptions {
+    commitSha: string;
+    targetRemoteRepo: string;
+    baseBranch: string;
+}
+
+export interface BranchAndPROptions extends GitOptions {
+    githubPAT: string;
+    dryRun: boolean;
+    newBranchName: string;
+    userName?: string;
+    email?: string;
+}
+
+// Logging utilities
+export function logWarning(message: string): void {
+    console.log(`##vso[task.logissue type=warning]${message}`);
+}
+
+export function logError(message: string): void {
+    console.log(`##vso[task.logissue type=error]${message}`);
+}
+
+export async function git(args: string[], printCommand = true): Promise<string> {
+    if (printCommand) {
+        console.log(`git ${args.join(' ')}`);
+    }
+
+    const git = spawnSync('git', args);
+    if (git.status !== 0) {
+        const err = git.stderr.toString();
+        if (printCommand) {
+            console.log(`Failed to execute git ${args.join(' ')}.`);
+        }
+        throw err;
+    }
+
+    const stdout = git.stdout.toString();
+    if (printCommand) {
+        console.log(stdout);
+    }
+    return stdout;
+}
+
+export async function getCommitFromNugetAsync(packageInfo: NugetPackageInfo): Promise<string | null> {
```

Comment:
can the createTags task also be updated to use this helper?
https://github.com/dotnet/vscode-csharp/blob/main/tasks/createTagsTasks.ts#L198